.git-hooks/pre-commit

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env node
+
+const { execSync } = require("child_process");
+const { existsSync, promises: fsPromises } = require("fs");
+const { join, resolve } = require("path");
+
+const excludedDirs = ["node_modules"];
```

Comment:
Do you guys think it's a good idea to make this code read and exclude whatever is in the `.gitignore` file, instead of hard coding `node_modules`, in case we have something else in the future that we don't want to format (since we don't need to format anything that is not going to be pushed)

Reply:
That is a good point, it should exclude anything in the `.gitignore` file.

Reply:
Yes guys, you are right. I was considering the same approach, as it offers more flexibility. 
However, considering that:
1. We are formatting `.js` files only
2. Among the directories/files listed in `.gitignore`, only the `node_modules` directory contains `.js` files.
3. After checking a list of most frequently "gitignored" files/directories, it seems unlikely that we will encounter other `.js` files within `.gitignore`.
4. The `.gitignore` file is typically not very dynamic once the project architecture is established.
5. The execution time using the hardcoded approach is significantly faster.

Taking all these factors into account, I hdecided to go for the hardcoded approach, as it strikes a suitable balance between speed and efficiency. However, if you still prefer to pursue the .gitignore approach, I would be glad to refactor the code accordingly